src/main/java/org/red5/server/net/rtmp/codec/RTMPProtocolDecoder.java

```diff
@@ -313,9 +316,16 @@ public Packet decodePacket(RTMPConnection conn, RTMPDecodeState state, IoBuffer
             state.bufferDecoding(headerLength + readAmount);
             return null;
         }
-        buf.put(in.array(), in.position(), readAmount);
-        //BufferUtils.put(buf, in, readAmount);
-        if (buf.position() < header.getSize()) {
+        int correction = 0;
+        //we will fill the buffer chunk by chunk skipping any CHUNK_DELIMITER found
+        for (int i = in.position(); i < in.position() + readAmount; i += CHUNK_SIZE) {
+        	if (in.array()[i] == CHUNK_DELIMITER) {
+        		i++;
+        		correction++;
+        	}
+    		buf.put(in.array(), i, Math.min(CHUNK_SIZE, in.position() + readAmount + correction - i));

```

Comment:
This part of code helps to remove 'C3' chunk delimiter bytes
the constant size here can easily be replaced with variable size